services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DeviceStatus.java

```diff
@@ -39,6 +37,7 @@
      *
      * @return {@code true} if this device was auto-provisioned.
      */
+    @JsonProperty(RegistryManagementConstants.FIELD_AUTO_PROVISIONED)
```

Comment:
this means that the property will *always* be included in the JSON document. Is that what you want to achieve?

Reply:
I moved that annotation to the getter, since I found out when doing manual testing for double-checking if the _authorities_ property is included (see https://github.com/eclipse/hono/pull/2094#discussion_r542625588) that it seems like this annotation has no effect when it is put on the field level, meaning the property is serialised using a camel-cased name.
Unfortunately I did not manage yet to find out what the root-cause for this behaviour is. There's even a unit test which is still succeeding when the property is on the field level. So I guess this must be caused by using Jackson together with Vert.x...

Besides that I think that the property being always serialised is not the effect of this annotation being put on the method level, but stems from the primitive type being returned by the getter (which we deliberately set to avoid null-checking).

Semantically I think it is correct: devices which were created before the implementation of auto-provisioning won't have the corresponding flag set resulting in returning false for the _auto-provisioned_ property. In my opinion this is correct as they actually haven't been auto-provisioned. 
I didn't see any value in exposing if the device was created before or after this feature was implemented to the client which is why I deemed it acceptable.